packages/flutter_tools/lib/src/android/java.dart

```diff
@@ -148,8 +148,8 @@ class Java {
     // Should look something like 'openjdk 19.0.2 2023-01-17'.
     final String longVersionText = versionLines.length >= 2 ? versionLines[1] : versionLines[0];
 
-    // The contents that matter come in the format '11.0.18' or '1.8.0_202'.
-    final RegExp jdkVersionRegex = RegExp(r'\d+\.\d+(\.\d+(?:_\d+)?)?');
+    // The contents that matter come in the format '11.0.18', '1.8.0_202 or 21'.
+    final RegExp jdkVersionRegex = RegExp(r'(?<version>\d+(\.\d+(\.\d+(?:_\d+)?)?)?)');
```

Comment:
If this no longer requires `\.`, then isn't the code a few lines further down essentially dead code now?

If we're going to relax the main check instead of making the check below allow `(java|openjdk)`, then it doesn't seem like we need the special codepath there at all any more.